Assets/Scripts/Combat/Projectile.cs

```diff
@@ -1,17 +1,29 @@
+using System.Collections.Generic;
+
 using UnityEngine;
 
 namespace CwispyStudios.TankMania.Combat
 {
+  using Poolers;
   using Stats;
   using Visuals;
 
   public class Projectile : MonoBehaviour
   {
-    // Particle system/vfx to play when projectile impacts
-    [SerializeField] private CFX_AutoDestructShuriken explosionVfx = null;
+    [Tooltip("Units without a soft match with these properties do weakened damage and do not do splash damage.")]
+    [SerializeField] private UnitType effectiveAgainst;
+    [Tooltip("Particle system/vfx to play when projectile impacts normally")]
```

Comment:
For debugging, it would be nice to be able for these fields to be left empty. Also perhaps should put all the VFX components in a separate scripts

Reply:
Resolved, added null check